pkg/querier/worker/frontend_processor_test.go

```diff
@@ -93,6 +94,57 @@ func TestFrontendProcessor_processQueriesOnSingleStream(t *testing.T) {
 		processClient.AssertNumberOfCalls(t, "Send", 1)
 	})
 }
+func TestFrontendProcessor_QueryTime(t *testing.T) {
+	runTest := func(t *testing.T, statsEnabled bool) {
+		fp, processClient, requestHandler := prepareFrontendProcessor()
+
+		recvCount := atomic.NewInt64(0)
+		queueTime := 3 * time.Second
+
+		processClient.On("Recv").Return(func() (*frontendv1pb.FrontendToClient, error) {
+			switch recvCount.Inc() {
+			case 1:
+				return &frontendv1pb.FrontendToClient{
+					Type:           frontendv1pb.HTTP_REQUEST,
+					HttpRequest:    nil,
+					QueueTimeNanos: queueTime.Nanoseconds(),
+					StatsEnabled:   statsEnabled,
+				}, nil
+			default:
+				// No more messages to process, so waiting until terminated.
+				<-processClient.Context().Done()
+				return nil, processClient.Context().Err()
+			}
+		})
+
+		workerCtx, workerCancel := context.WithCancel(context.Background())
+
+		requestHandler.On("Handle", mock.Anything, mock.Anything).Run(func(args mock.Arguments) {
+			workerCancel()
+
+			stat := stats.FromContext(args.Get(0).(context.Context))
+
+			if statsEnabled {
+				require.Equal(t, queueTime, stat.LoadQueueTime())
+			} else {
+				require.Equal(t, time.Duration(0), stat.LoadQueueTime())
+			}
+		}).Return(&httpgrpc.HTTPResponse{}, nil)
+
+		fp.processQueriesOnSingleStream(workerCtx, nil, "127.0.0.1")
+
+		// We expect Send() to be called once, to send the query result.
+		processClient.AssertNumberOfCalls(t, "Send", 1)
+	}
+
+	t.Run("query stats enabled should record query time", func(t *testing.T) {
```

Comment:
query time or queue time?

Reply:
Fixed in https://github.com/grafana/mimir/pull/6537/commits/0c1786c77965bfd62692ae09efce2beb13f54422.